modules/internal/DefaultRectangular.chpl

```diff
@@ -848,32 +1294,127 @@ module DefaultRectangular {
+                               true /* ignoreRunning */,
+                               1 /* minIndicesPerTask */,
+                               dom.ranges);
+          if numChunks == 0 {
+            mdParDim = 1;
+            mdNumChunks = 1;
+          } else {
+            mdParDim = parDim;
+            mdNumChunks = numChunks;
+          }
+        }
+        mdRLo = dom.dsiDim(mdParDim).alignedLow;
+        mdRHi = dom.dsiDim(mdParDim).alignedHigh;
+        mdRLen = dom.dsiDim(mdParDim).length;
+        dataVec = _ddata_allocate(_ddata(eltType), mdNumChunks);
+        for i in 0..#mdNumChunks {
+          const (lo, hi) = mdChunk2Ind(i);
+          const chunkSize = if mdRLen == 0 then 0
+                            else size / mdRLen * (hi - lo + 1);
+          dataVec(i) = _ddata_allocate(eltType, chunkSize);
+        }
+      }
+
       initShiftedData();
       if rank == 1 && !stridable then
         dataAllocRange = dom.dsiDim(1);
+    }
+
+    inline proc mdInd2Chunk(ind, mdnc=mdNumChunks,
+                            mdrlo=mdRLo, mdrlen=mdRLen, str=1:idxType)
+      where !defRectSimpleDData {
+      if stridable then
+        return ((((ind - mdrlo) / str) * mdnc:idxType) / mdrlen):int;
+      else
+        return (((ind - mdrlo) * mdnc:idxType) / mdrlen):int;
+    }
 
+    inline proc mdChunk2Ind(chunk, mdnc=mdNumChunks,
+                            mdrlo=mdRLo, mdrhi=mdRHi, mdrlen=mdRLen,
+                            str=1:idxType)
+      where !defRectSimpleDData {
+      if stridable {
+        const (lo, hi) = _computeBlock(mdrlen, mdnc, chunk, mdrhi - mdrlo, 0, 0);
+        return (lo * str + mdrlo, hi * str + mdrlo);
+      } else {
+        return _computeBlock(mdrlen, mdnc, chunk, mdrhi, mdrlo, mdrlo);
+      }
     }
 
-    inline proc getDataIndex(ind: idxType ...1) where rank == 1
-      return getDataIndex(ind);
+    inline proc getDataIndex(ind: idxType ...1,
+                             param getShifted = true,
+                             param getChunked = !defRectSimpleDData)
+      where rank == 1
+      return getDataIndex(ind, getShifted=getShifted, getChunked=getChunked);
+
+    inline proc getDataIndex(ind: rank*idxType,
+                             param getShifted = true,
+                             param getChunked = !defRectSimpleDData) {
+      param chunkify = !defRectSimpleDData && getChunked;
+
+      // gbt TODO: this str division may occur even when const 1
+      inline proc chunked_dataIndex(sum, str) {
+        if mdNumChunks == 1 {
+          return (0, sum);
+        } else {
+          const mdpd = mdParDim;
+          const mdnc = mdNumChunks;
+          const mdrlo = mdRLo;
+          const mdrhi = mdRHi;
+          const mdrlen = mdRLen;
+          const chunk = mdInd2Chunk(ind(mdpd), mdnc, mdrlo, mdrlen, str);
+          const (lo, _) = mdChunk2Ind(chunk, mdnc, mdrlo, mdrhi, mdrlen);
+          return (chunk, sum - (lo - mdrlo) * blk(mdpd) / str);
```

Comment:
I originally did this to reduce the number of GETs for references to non-distributed remote arrays.  Based on a little bit of new testing, it now looks like this is unnecessary, so I'll undo this.